src/panels/InertiaPanel.php

```
@@ -0,0 +1,303 @@
+
+        Event::on(
+            View::class,
+            View::EVENT_BEFORE_RENDER,
+            function (ViewEvent $event): void {
+                $page = $event->params['page'] ?? null;
+
+                if ($page instanceof JsonSerializable && is_a($page, self::PAGE_CLASS)) {
+                    $this->page = $page;
+                }
+            },
+        );
```

PHPMD (2.15.0)

[error] Avoid using static access to class '\yii\base\Event' in method 'init'. (undefined)

(StaticAccess)